platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/RepositoryChainComponentMetaDataResolver.java

```diff
@@ -59,7 +65,9 @@ public void resolve(ComponentIdentifier identifier, ComponentOverrideMetadata co
             throw new UnsupportedOperationException("Can resolve meta-data for module components only.");
         }
 
-        resolveModule((ModuleComponentIdentifier) identifier, componentOverrideMetadata, result);
+        metadataValueContainerCache.getReference(toDisplayName(identifier), () -> resolveModule((ModuleComponentIdentifier) identifier, componentOverrideMetadata))
```

Comment:
🤔 It's a little weird to use the DisplayName as the key vs the component identifier.